app/controllers/cases_controller.rb

```diff
@@ -0,0 +1,71 @@
+class CasesController < ApplicationController
+  before_action :set_case, only: [:show,:edit,:update,:destroy]
+
+  def index
+    @cases=Case.all
+  end
+
+  def show
+  end
+
+  def update
+    if @case.update(case_params)
+     redirect_to @case
+    else
+      render :edit 
+    end
+  end
+
+  def delete_document
+    case_id = Document.find(params[:document]).case_id
+    Document.destroy(params[:document])
+    redirect_to edit_case_path(Case.find(case_id)) 
+  end
+  
+  def edit
+    @documents = @case.documents
+  end
+
+  def new
+    @case = Case.new
+    @documents = @case.documents
```

Comment:
In edit action you have same code `@documents = @case.documents` create  a method for it and use it in before_action for edit and new action. We should remove duplicate code.